### ISSUE TITLE
Export object along with its dependencies

### ISSUE THREAD

**Issue by larshp** [larshp/abapGit#639]
If the developments in a system is not organized into packages, i.e. everything is in one package. Would like to export report ZFOOBAR to a different system, problem is that ZFOOBAR uses a lot of different objects, it could be nice if abapGit would find these objects automatically and suggest which ones to export.

discussed briefly in https://github.com/larshp/abapGit/issues/544

Been writing some test code at https://github.com/larshp/dependencies, still a long way to go

**Comment by sbcgua** [comment]
I think there was some discussion to mention depencies (other AG packages) in `.abapgit`. Is this issue related ? (And are there any plans to implement this feature too in any near future ?) :)

**Comment by larshp** [comment]
https://github.com/larshp/dependencies got a long way last weekend, but there are a lot of bugs and stuff that is not handled yet. The idea is to have a checkmark in eg. Advanced -> Object to Files, which will trigger all dependencies(except SAP standard) to be downloaded.

Do you need this feature? First step would be to merge the stuff from https://github.com/larshp/dependencies and add some way to test/visualize this in abapGit

https://github.com/larshp/abapGit/issues/297 is "crazy", and more related to being able to do an offline syntax check(not sure how this will work out), but implementing this issue/feature first will help

**Comment by sbcgua** [comment]
Not that I personally really need it. Maybe I'd like to split my zmockup_loader into 2 packages (parser and mockup stuff). But I also think it would be a good feature for future. Kind of package.json ...and then maybe make some npm-like site someday ... 
also my colleague is working on that announcement feature currently and we've got an idea that actually any package can have changes log (under certain rules) which can be displayed in AG interface universally. And all these features are kind of play together well as a package management system ... just thoughts ...

**Comment by larshp** [comment]
requirements to other packages: https://github.com/larshp/abapGit/issues/217

the scope for this issue is to find dependencies within the system, eg. which objects are used by a program

**Comment by gregorwolf** [comment]
Hi @RainerWinkler could your new developments in [SAP2Moose](https://github.com/SAP2Moose/SAP2Moose) help to solve this issue?

**Comment by RainerWinkler** [comment]
Hi @gregorwolf 
this sounds like what I call down search. To find objects that are used by a coding. I added a down search to SAP2Moose recently (Issue https://github.com/SAP2Moose/SAP2Moose/issues/93 and later). The logic is not trivial and my implementation is far from complete.
I looked into https://github.com/larshp/dependencies, there are a lot of similarities with the coding I implemented for downsearch in SAP2Moose.
It will definitively help to exchange knowledge about testcases and implementation details (Or more when this is possible)
FYI @larshp
